renderer/renderer.go

```diff
@@ -315,11 +328,43 @@ func RenderGemini(s string, width int, proxied bool) (string, []string) {
 	pre := false
 	buf := "" // Block of regular or preformatted lines
 
+	// Language, formatter, and style for syntax highlighting
+	lang := ""
+	formatterName := TermColor
+	styleName := viper.GetString("a-general.highlight_style")
+
 	// processPre is for rendering preformatted blocks
 	processPre := func() {
 
+		// Perform syntax highlighting if language is set
+		if lang != "" {
+			style := styles.Get(styleName)
+			if style == nil {
+				style = styles.Fallback
+			}
+			formatter := formatters.Get(formatterName)
+			if formatter == nil {
+				formatter = formatters.Fallback
+			}
+			lexer := lexers.Get(lang)
+			if lexer == nil {
+				lexer = lexers.Fallback
+			}
+
+			// Tokenize and format the text after stripping ANSI codes, replacing buffer if there are no errors
+			iterator, err := lexer.Tokenise(nil, ansiRegex.ReplaceAllString(buf, ""))
+			if err == nil {
+				formattedBuffer := new(bytes.Buffer)
+				if formatter.Format(formattedBuffer, style, iterator) == nil {
+					// Strip extra newline added by Chroma and replace buffer
+					buf = string(trailingNewline.ReplaceAll(formattedBuffer.Bytes(), []byte{}))
+				}
+			}
+		}
+
 		// Support ANSI color codes in preformatted blocks - see #59
-		if viper.GetBool("a-general.color") && viper.GetBool("a-general.ansi") {
+		// This will also execute if code highlighting language is set
+		if viper.GetBool("a-general.color") && (viper.GetBool("a-general.ansi") || lang != "") {
```

Comment:
There needs to be multiple if statements or something here, to support a configuration where the user doesn't want ANSI rendered, but they do want code highlighted, or vice versa.

Reply:
OK, that makes sense, I was thinking of it more from the perspective of "highlighting uses ANSI, so using highlighting requires ANSI" but it probably does make more sense to think of them as two separate features... people may want code highlighting without arbitrary colorful preformatted text.

Reply:
Ok, actually this is almost correct. So this only executes on a block if ANSI highlighting is on OR if a language is detected, but instead it should execute if ANSI highlighting is on OR if the syntax was *successfully tokenized* based on the detected language. This requires a new boolean variable that is set if err == nil from the call to lexer.Tokenise().

Reply:
I was misunderstanding before I think, but thanks for catching that bug as well. Looks good now.